### DISCUSSION TITLE
Enforce proper contrast for button text & icons

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5806] [Feature Requests]
<img width="1097" alt="image" src="https://github.com/specify/specify7/assets/37256050/bb1ccca6-9471-4f51-a627-ed6c19f776bb">

This is something that is particularly important once we implement #3241. You can select a dark color for buttons and the text on the button can lose the content necessary to make it legible.

Icons & text on dark backgrounds should become white, such as in this case. Specify should be smart enough to make this happen!

<img width="549" alt="image" src="https://github.com/specify/specify7/assets/37256050/f0050d02-a988-4105-972f-715ba9d0af2e">

**Comment by maxpatiiuk** [comment]
This will be easy to address once CSS adds support for automatic text color:

> A new CSS feature is currently being developed and will help address these concerns.
Keep an eye on it and integrate it once it is ready to be used (https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-contrast)

See https://github.com/specify/specify7/issues/2615#issuecomment-1538610878

Until then, our best bet is to find a library for this

**Comment by grantfitzsimmons** [comment]
@maxpatiiuk Should this not be a blocker for #3241? It seems within the scope of that change. When customizing colors, you'll find this problematic very quickly.

**Comment by grantfitzsimmons** [comment]
After researching online, there are a lot of solutions. We just need to pick one and implement it for the time being.

**Comment by maxpatiiuk** [comment]
I would call this a necessary nice extra on top of https://github.com/specify/specify7/pull/3241 to make it more usable

Given that this is not an urgent and not high importance, it's fine to delay that pull request to implement this (except for the fact that we have too many pull requests already and developers are having to waste a ton of time with merge conflicts at the moment)

Given that, I think we should release https://github.com/specify/specify7/pull/3241 anyway and don't advertise it until this is fixed

**Comment by grantfitzsimmons** [comment]
From https://github.com/specify/specify7/pull/3470#issuecomment-1550279637
> > This is to prevent setting the logo to an inappropriate color - but we could remove this prevention if you want
> 
> Dark (dark purple), "inappropriate" colors are OK, light (neon green) "inappropriate" colors are not. Can we make a distinction?
